polkadot/xcm/src/v3/mod.rs

```diff
@@ -60,13 +63,28 @@ pub const VERSION: super::Version = 3;
 /// An identifier for a query.
 pub type QueryId = u64;
 
-#[derive(Derivative, Default, Encode, Decode, TypeInfo)]
+#[derive(Derivative, Default, Encode, TypeInfo)]
 #[derivative(Clone(bound = ""), Eq(bound = ""), PartialEq(bound = ""), Debug(bound = ""))]
 #[codec(encode_bound())]
-#[codec(decode_bound())]
 #[scale_info(bounds(), skip_type_params(Call))]
 pub struct Xcm<Call>(pub Vec<Instruction<Call>>);
 
+const MAX_INSTRUCTIONS_TO_DECODE: u32 = 100;
+const TOO_MANY_INSTRUCTIONS_ERROR_MESSAGE: &'static str = "Too many instructions to decode";
+
+impl<Call> Decode for Xcm<Call> {
+	// Taken from `BoundedVec`'s `Decode` implementation
+	// TODO (v4): Use `BoundedVec`
+	fn decode<I: CodecInput>(input: &mut I) -> core::result::Result<Self, CodecError> {
+		let len: u32 = <Compact<u32>>::decode(input)?.into();
+		if len > MAX_INSTRUCTIONS_TO_DECODE {
+			return Err(TOO_MANY_INSTRUCTIONS_ERROR_MESSAGE.into())
+		}
+		let decoded_instructions = decode_vec_with_len(input, len as usize)?;
+		Ok(Self(decoded_instructions))
```

Comment:
Or you do `BoundedVec::<Instruction, ConstU32<MAX_INSTRUCTIONS_TO_DECODE>>::decode(…` and then access `into_inner`.  
But its pretty equivalent.